gpcontrib/pg_hint_plan/pg_hint_plan.c

```diff
@@ -5038,12 +5031,11 @@ void plpgsql_query_erase_callback(ResourceReleasePhase phase,
 #include "pg_stat_statements.c"
 
 
-#ifdef USE_ORCA
 /*
  * This function hook allows external code (i.e. backend) to parse a query into
  * hint structures.
  */
-static void *
+static HintState *
```

Comment:
Why did you not remove cast to (HintState *)  in https://github.com/arenadata/gpdb/blob/fb35e05f2e77c27a4e8f8ef50fa60563c94cb8f3/src/backend/gpopt/utils/COptTasks.cpp#L573 ?